### ISSUE TITLE
Restore checkpoint on desync blockchain state

### ISSUE THREAD

**Issue by h4sh3d** [farcaster-project/farcaster-node#690]
## Scenario

Initiated a swap against farcaster.dev as Bob (I funded btc), I locked btc, other locked xmr, I saw it and sent the buy message, other bought

```
$ docker-compose exec farcasterd swap-cli progress 0xc9e80d6dfbd0cdf12d41c6d7e31714013877494f374f6e3150ee14f16abc1ccf --follow
progress: Proposing to take swap 0xc9e8…1ccf to Maker remote peer
progress: BobState(Start) -> BobState(Commit)
progress: BobState(Commit) -> BobState(Reveal)
progress: BobState(Reveal) -> BobState(CoreArb)
progress: BobState(CoreArb) -> BobState(BuySig)
```

Then my bitcoin syncer stopped working (no more blocks seen and no tx seen).

I killed and restarted farcaster, restored the checkpoint:

```
[2022-08-23T14:11:29Z INFO  farcaster_node::farcasterd::runtime] Restoring swap 0xc9e8…1ccf
...
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::runtime] 0xc9e8…1ccf | checkpointing bob pre buy swapd state
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::swap_state] checkpoint bob pre buy already set
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::runtime] subscribing with syncer for receiving raw buy tx
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Buy transaction (66e487ed4ca91395fbb056529fc3c3ecf334cab33d8c24d881ffcb9828e66558)
...
[2022-08-23T14:11:49Z WARN  farcaster_node::swapd::runtime] expected BobState(BuySigB), found BobState(CoreArb). Any chance you reused the destination/refund address in the cli command? For your own privacy, do not reuse bitcoin addresses. Txid 66e487ed4ca91395fbb056529fc3c3ecf334cab33d8c24d881ffcb9828e66558
...
[2022-08-23T14:12:10Z INFO  farcaster_node::swapd::runtime] 0xc9e8…1ccf | State transition: BobState(CoreArb) -> BobState(BuySig)
```

## Full logs

```
[2022-08-23T14:10:41Z INFO  farcaster_node::config] Loading config file at: /var/lib/farcaster/farcasterd.toml
[2022-08-23T14:10:41Z DEBUG farcaster_node::farcasterd::runtime] Launching walletd as a separate process using `/usr/local/bin/walletd` as binary
[2022-08-23T14:10:41Z INFO  farcaster_node::farcasterd::runtime] tor opts: None
[2022-08-23T14:10:41Z DEBUG farcaster_node::farcasterd::runtime] Executing `"/usr/local/bin/walletd" "-d" "~/.farcaster" "-m" "{data_dir}/msg" "-x" "{data_dir}/ctl" "--token" "9b90a5c3020a31b5ebfc37ceea219a74"`
[2022-08-23T14:10:41Z DEBUG farcaster_node::farcasterd::runtime] Launching databased as a separate process using `/usr/local/bin/databased` as binary
[2022-08-23T14:10:41Z INFO  farcaster_node::farcasterd::runtime] tor opts: None
[2022-08-23T14:10:41Z DEBUG farcaster_node::farcasterd::runtime] Executing `"/usr/local/bin/databased" "-d" "~/.farcaster" "-m" "{data_dir}/msg" "-x" "{data_dir}/ctl"`
[2022-08-23T14:10:41Z INFO  farcaster_node::service] New service farcasterd started
[2022-08-23T14:10:42Z INFO  farcaster_node::service] New service walletd started
[2022-08-23T14:10:42Z INFO  farcaster_node::farcasterd::runtime] Service walletd is now connected
[2022-08-23T14:10:42Z INFO  farcaster_node::service] New service databased started
[2022-08-23T14:10:42Z INFO  farcaster_node::farcasterd::runtime] Service databased is now connected
[2022-08-23T14:11:27Z INFO  farcaster_node::farcasterd::runtime] launching syncer with: ["--blockchain", "Bitcoin", "--network", "Testnet", "--electrum-server", "ssl://blockstream.info:993"]
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Launching syncerd as a separate process using `/usr/local/bin/syncerd` as binary
[2022-08-23T14:11:27Z INFO  farcaster_node::farcasterd::runtime] tor opts: None
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Executing `"/usr/local/bin/syncerd" "-d" "~/.farcaster" "-m" "{data_dir}/msg" "-x" "{data_dir}/ctl" "--blockchain" "Bitcoin" "--network" "Testnet" "--electrum-server" "ssl://blockstream.info:993"`
[2022-08-23T14:11:27Z INFO  farcaster_node::farcasterd::runtime] launching syncer with: ["--blockchain", "Monero", "--network", "Testnet", "--monero-daemon", "http://stagenet.community.rino.io:38081", "--monero-rpc-wallet", "http://walletrpc:38083"]
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Launching syncerd as a separate process using `/usr/local/bin/syncerd` as binary
[2022-08-23T14:11:27Z INFO  farcaster_node::farcasterd::runtime] tor opts: None
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Executing `"/usr/local/bin/syncerd" "-d" "~/.farcaster" "-m" "{data_dir}/msg" "-x" "{data_dir}/ctl" "--blockchain" "Monero" "--network" "Testnet" "--monero-daemon" "http://stagenet.community.rino.io:38081" "--monero-rpc-wallet" "http://walletrpc:38083"`
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Launching swapd as a separate process using `/usr/local/bin/swapd` as binary
[2022-08-23T14:11:27Z INFO  farcaster_node::farcasterd::runtime] tor opts: None
[2022-08-23T14:11:27Z DEBUG farcaster_node::farcasterd::runtime] Executing `"/usr/local/bin/swapd" "-d" "~/.farcaster" "-m" "{data_dir}/msg" "-x" "{data_dir}/ctl" "c9e80d6dfbd0cdf12d41c6d7e31714013877494f374f6e3150ee14f16abc1ccf" "Offer:Cke4ftrP5A7HwaFvWnqcoyW7oYhXzpRcV1LQM2fvVdFMNR4h47P6iGo11111uNZ1adYM7a4LW1H5H11111TBPSF1113GTvtvqfD1111112g5B8cqRMKpcSttoC2RGdaKmhaxcBufKUapvpCcKVpVgCzp4H111111111111111111111111111111111111111115ymH83buC2CjHJ3u" "Taker"`
[2022-08-23T14:11:27Z INFO  farcaster_node::syncerd::runtime] Creating new Monero (Testnet) syncer
[2022-08-23T14:11:27Z INFO  farcaster_node::syncerd::runtime] Creating new Bitcoin (Testnet) syncer
[2022-08-23T14:11:27Z INFO  farcaster_node::syncerd::monero_syncer] monero syncer servers: MoneroSyncerServers { monero_daemon: "http://stagenet.community.rino.io:38081", monero_rpc_wallet: "http://walletrpc:38083", monero_lws: None }
[2022-08-23T14:11:27Z DEBUG farcaster_node::syncerd::bitcoin_syncer] bitcoin synclet using proxy: None
[2022-08-23T14:11:27Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating ElectrumRpc client
[2022-08-23T14:11:27Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating ElectrumRpc client
[2022-08-23T14:11:28Z INFO  farcaster_node::swapd::runtime] Starting swap: 0xc9e80d6dfbd0cdf12d41c6d7e31714013877494f374f6e3150ee14f16abc1ccf
[2022-08-23T14:11:28Z INFO  farcaster_node::swapd::runtime] 0xc9e8…1ccf | Initial state: BobState(Start)
[2022-08-23T14:11:28Z DEBUG farcaster_node::syncerd::bitcoin_syncer] New ElectrumRpc at height 2343545
[2022-08-23T14:11:28Z DEBUG farcaster_node::syncerd::bitcoin_syncer] New ElectrumRpc at height 2343545
[2022-08-23T14:11:28Z INFO  farcaster_node::syncerd::syncer_state] Bitcoin incremented height 2343545
[2022-08-23T14:11:28Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating ElectrumRpc client
[2022-08-23T14:11:28Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating fee polling electrum client
[2022-08-23T14:11:28Z DEBUG farcaster_node::syncerd::bitcoin_syncer] New ElectrumRpc at height 2343545
[2022-08-23T14:11:28Z INFO  farcaster_node::service] New service Bitcoin (Testnet) syncer started
[2022-08-23T14:11:28Z INFO  farcaster_node::service] New service Monero (Testnet) syncer started
[2022-08-23T14:11:28Z INFO  farcaster_node::farcasterd::runtime] Service Monero (Testnet) syncer is now connected
[2022-08-23T14:11:28Z INFO  farcaster_node::farcasterd::runtime] Syncer Monero (Testnet) syncer is registered; total 1 syncers are known
[2022-08-23T14:11:28Z WARN  farcaster_node::farcasterd::runtime] syncer pair not ready
[2022-08-23T14:11:28Z DEBUG farcaster_node::farcasterd::runtime] Daemon Monero (Testnet) syncer is known: we spawned it to create a new peer connection by a request from farcasterd
[2022-08-23T14:11:28Z INFO  farcaster_node::farcasterd::runtime] Service Bitcoin (Testnet) syncer is now connected
[2022-08-23T14:11:28Z INFO  farcaster_node::farcasterd::runtime] Syncer Bitcoin (Testnet) syncer is registered; total 2 syncers are known
[2022-08-23T14:11:28Z DEBUG farcaster_node::farcasterd::runtime] Daemon Bitcoin (Testnet) syncer is known: we spawned it to create a new peer connection by a request from farcasterd
[2022-08-23T14:11:29Z INFO  farcaster_node::service] New service swap<0xc9e8…1ccf> started
[2022-08-23T14:11:29Z INFO  farcaster_node::farcasterd::runtime] Service swap<0xc9e8…1ccf> is now connected
[2022-08-23T14:11:29Z INFO  farcaster_node::farcasterd::runtime] Restoring swap 0xc9e8…1ccf
[2022-08-23T14:11:29Z INFO  farcaster_node::farcasterd::runtime] Swap 0xc9e8…1ccf is registered; total 1 swaps are known
[2022-08-23T14:11:29Z INFO  farcaster_node::syncerd::syncer_state] Monero incremented height 1163765
[2022-08-23T14:11:29Z INFO  farcaster_node::walletd::runtime] Restoring wallet for swap 0xc9e8…1ccf
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::runtime] 0xc9e8…1ccf | Restoring swap
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Cancel transaction (16d89f9ae3e9fb8a38855b84cd0f7c822660f52d31d2e2eb6d604750c3410749)
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Refund transaction (0d9e6f06bbce9ab1193e363293643ed349d00b4b4746e2da3c9aea380adcb165)
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Arbitrating Lock transaction (361dc69c49393e3e58739ad4d1480ad03e5c7efb84c734494a06896320781df3)
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Address's secret view key for Accordant Lock: d68bb283f10b233fbf4eccd5f42eefa2b50ecde7108097f22a25e21cb163db03
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Address's public spend key for Accordant Lock: c8c648c72aeea99b4bb61c96a990fa1942db302e75c8378b15659c6b317c3938
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Accordant Lock on address 59SSBMEwzNcSyZftCbkqk55E4mDHUDEpnQGHXdEzaECGATD52kV1R5Fb28pXR2jmjB5TLngkWBb7u2eGw8NKJaXr6pNy7Fw
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::runtime] 0xc9e8…1ccf | checkpointing bob pre buy swapd state
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::swap_state] checkpoint bob pre buy already set
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::runtime] subscribing with syncer for receiving raw buy tx
[2022-08-23T14:11:29Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Buy transaction (66e487ed4ca91395fbb056529fc3c3ecf334cab33d8c24d881ffcb9828e66558)
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] received new task: WatchTransaction { id: TaskId(3), lifetime: 18446744073709551615, hash: [73, 7, 65, 195, 80, 71, 96, 109, 235, 226, 210, 49, 45, 245, 96, 38, 130, 124, 15, 205, 132, 91, 133, 56, 138, 251, 233, 227, 154, 159, 216, 22], confirmation_bound: 50000 }
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] received new task: WatchTransaction { id: TaskId(4), lifetime: 18446744073709551615, hash: [101, 177, 220, 10, 56, 234, 154, 60, 218, 226, 70, 71, 75, 11, 208, 73, 211, 62, 100, 147, 50, 54, 62, 25, 177, 154, 206, 187, 6, 111, 158, 13], confirmation_bound: 50000 }
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] received new task: WatchTransaction { id: TaskId(5), lifetime: 18446744073709551615, hash: [243, 29, 120, 32, 99, 137, 6, 74, 73, 52, 199, 132, 251, 126, 92, 62, 208, 10, 72, 209, 212, 154, 115, 88, 62, 62, 57, 73, 156, 198, 29, 54], confirmation_bound: 50000 }
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] trying to broadcast tx: "02000000000101379149ae56567cbc3765a79db9f443f4246787784b0dddd75d1c031f2388764001000000000000008001dc050000000000002200202637126fd5b562a3a86d33aa7271ff74c6fd8fc8d36b2b8ca8dfe9193ca9f92102483045022100c6a20cc534004c95d55dd913bad6129cd73499827f0d3a725c85b52bf765179d022038973b331beb7b0b557a9261a1166f382f876dd1f9f8b0d945506a4e1b020ccd012103d2f142863f2329785f2007461533f55717bb16e5958310dbf1ccb61cfa50b9e200000000"
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] received new task: WatchTransaction { id: TaskId(8), lifetime: 18446744073709551615, hash: [88, 101, 230, 40, 152, 203, 255, 129, 216, 36, 140, 61, 179, 202, 52, 243, 236, 195, 195, 159, 82, 86, 176, 251, 149, 19, 169, 76, 237, 135, 228, 102], confirmation_bound: 50000 }
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating transaction broadcast electrum client
[2022-08-23T14:11:29Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, HeightChanged(HeightChanged { id: TaskId(1), block: [177, 116, 17, 250, 63, 70, 83, 195, 21, 86, 220, 20, 7, 195, 81, 44, 231, 216, 178, 173, 22, 66, 72, 30, 30, 0, 0, 0, 0, 0, 0, 0], height: 2343545 }))
[2022-08-23T14:11:29Z DEBUG farcaster_node::swapd::syncer_client] Bitcoin new height 2343545
[2022-08-23T14:11:30Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, HeightChanged(HeightChanged { id: TaskId(2), block: [191, 162, 17, 66, 202, 236, 62, 247, 89, 30, 252, 233, 134, 163, 105, 110, 33, 184, 216, 208, 221, 207, 56, 51, 36, 81, 47, 73, 1, 137, 227, 131], height: 1163765 }))
[2022-08-23T14:11:30Z DEBUG farcaster_node::swapd::syncer_client] Monero new height 1163765
[2022-08-23T14:11:30Z ERROR farcaster_node::syncerd::bitcoin_syncer] failed to broadcast tx: Electrum server error: "sendrawtransaction RPC error: {\"code\":-25,\"message\":\"bad-txns-inputs-missingorspent\"}"
[2022-08-23T14:11:30Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionBroadcasted(TransactionBroadcasted { id: TaskId(6), tx: [2, 0, 0, 0, 0, 1, 1, 55, 145, 73, 174, 86, 86, 124, 188, 55, 101, 167, 157, 185, 244, 67, 244, 36, 103, 135, 120, 75, 13, 221, 215, 93, 28, 3, 31, 35, 136, 118, 64, 1, 0, 0, 0, 0, 0, 0, 0, 128, 1, 220, 5, 0, 0, 0, 0, 0, 0, 34, 0, 32, 38, 55, 18, 111, 213, 181, 98, 163, 168, 109, 51, 170, 114, 113, 255, 116, 198, 253, 143, 200, 211, 107, 43, 140, 168, 223, 233, 25, 60, 169, 249, 33, 2, 72, 48, 69, 2, 33, 0, 198, 162, 12, 197, 52, 0, 76, 149, 213, 93, 217, 19, 186, 214, 18, 156, 215, 52, 153, 130, 127, 13, 58, 114, 92, 133, 181, 43, 247, 101, 23, 157, 2, 32, 56, 151, 59, 51, 27, 235, 123, 11, 85, 122, 146, 97, 161, 22, 111, 56, 47, 135, 109, 209, 249, 248, 176, 217, 69, 80, 106, 78, 27, 2, 12, 205, 1, 33, 3, 210, 241, 66, 134, 63, 35, 41, 120, 95, 32, 7, 70, 21, 51, 245, 87, 23, 187, 22, 229, 149, 131, 16, 219, 241, 204, 182, 28, 250, 80, 185, 226, 0, 0, 0, 0], error: Some("failed to broadcast tx: Electrum server error: \"sendrawtransaction RPC error: {\\\"code\\\":-25,\\\"message\\\":\\\"bad-txns-inputs-missingorspent\\\"}\"") }))
[2022-08-23T14:11:32Z DEBUG farcaster_node::syncerd::monero_syncer] wallet doesn't exist, generating a new wallet: Server error: Failed to open wallet
[2022-08-23T14:11:36Z DEBUG farcaster_node::syncerd::monero_syncer] Watch wallet opened successfully
[2022-08-23T14:11:39Z DEBUG farcaster_node::syncerd::monero_syncer] Watch wallet opened successfully
[2022-08-23T14:11:39Z DEBUG farcaster_node::syncerd::bitcoin_syncer] Updated tx: 361dc69c49393e3e58739ad4d1480ad03e5c7efb84c734494a06896320781df3
[2022-08-23T14:11:40Z DEBUG farcaster_node::syncerd::bitcoin_syncer] Updated tx: 66e487ed4ca91395fbb056529fc3c3ecf334cab33d8c24d881ffcb9828e66558
[2022-08-23T14:11:43Z DEBUG farcaster_node::syncerd::monero_syncer] Watch wallet opened successfully
[2022-08-23T14:11:47Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(3), block: [0], confirmations: None, tx: [] }))
[2022-08-23T14:11:47Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(5), block: [126, 184, 201, 36, 68, 184, 5, 232, 7, 210, 208, 73, 31, 194, 50, 189, 116, 148, 51, 56, 26, 131, 30, 175, 236, 17, 0, 0, 0, 0, 0, 0], confirmations: Some(3), tx: [2, 0, 0, 0, 0, 1, 1, 55, 145, 73, 174, 86, 86, 124, 188, 55, 101, 167, 157, 185, 244, 67, 244, 36, 103, 135, 120, 75, 13, 221, 215, 93, 28, 3, 31, 35, 136, 118, 64, 1, 0, 0, 0, 0, 0, 0, 0, 128, 1, 220, 5, 0, 0, 0, 0, 0, 0, 34, 0, 32, 38, 55, 18, 111, 213, 181, 98, 163, 168, 109, 51, 170, 114, 113, 255, 116, 198, 253, 143, 200, 211, 107, 43, 140, 168, 223, 233, 25, 60, 169, 249, 33, 2, 72, 48, 69, 2, 33, 0, 198, 162, 12, 197, 52, 0, 76, 149, 213, 93, 217, 19, 186, 214, 18, 156, 215, 52, 153, 130, 127, 13, 58, 114, 92, 133, 181, 43, 247, 101, 23, 157, 2, 32, 56, 151, 59, 51, 27, 235, 123, 11, 85, 122, 146, 97, 161, 22, 111, 56, 47, 135, 109, 209, 249, 248, 176, 217, 69, 80, 106, 78, 27, 2, 12, 205, 1, 33, 3, 210, 241, 66, 134, 63, 35, 41, 120, 95, 32, 7, 70, 21, 51, 245, 87, 23, 187, 22, 229, 149, 131, 16, 219, 241, 204, 182, 28, 250, 80, 185, 226, 0, 0, 0, 0] }))
[2022-08-23T14:11:47Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(4), block: [0], confirmations: None, tx: [] }))
[2022-08-23T14:11:47Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Cancel not on the mempool
[2022-08-23T14:11:47Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(8), block: [177, 116, 17, 250, 63, 70, 83, 195, 21, 86, 220, 20, 7, 195, 81, 44, 231, 216, 178, 173, 22, 66, 72, 30, 30, 0, 0, 0, 0, 0, 0, 0], confirmations: Some(1), tx: [2, 0, 0, 0, 0, 1, 1, 243, 29, 120, 32, 99, 137, 6, 74, 73, 52, 199, 132, 251, 126, 92, 62, 208, 10, 72, 209, 212, 154, 115, 88, 62, 62, 57, 73, 156, 198, 29, 54, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 148, 4, 0, 0, 0, 0, 0, 0, 22, 0, 20, 154, 170, 80, 193, 126, 110, 61, 188, 123, 50, 133, 225, 94, 54, 215, 227, 147, 83, 205, 205, 3, 72, 48, 69, 2, 33, 0, 141, 179, 130, 157, 136, 124, 20, 245, 92, 0, 174, 216, 128, 37, 234, 189, 216, 9, 236, 208, 24, 125, 37, 149, 26, 24, 210, 225, 182, 141, 154, 75, 2, 32, 112, 28, 115, 52, 33, 155, 128, 156, 161, 73, 24, 157, 29, 139, 71, 66, 171, 149, 230, 9, 86, 76, 4, 186, 11, 233, 108, 60, 71, 174, 30, 215, 1, 72, 48, 69, 2, 33, 0, 189, 174, 180, 48, 239, 166, 38, 74, 4, 207, 48, 234, 103, 4, 63, 201, 72, 215, 122, 14, 3, 162, 33, 94, 75, 198, 137, 169, 28, 136, 51, 212, 2, 32, 13, 34, 249, 222, 112, 238, 167, 200, 130, 64, 158, 172, 0, 187, 5, 28, 169, 229, 9, 221, 56, 85, 9, 101, 71, 177, 194, 109, 167, 222, 20, 184, 1, 70, 33, 3, 39, 1, 85, 101, 47, 218, 34, 211, 101, 148, 64, 81, 148, 122, 210, 135, 27, 135, 104, 237, 65, 233, 248, 181, 64, 18, 53, 25, 20, 218, 45, 92, 173, 33, 2, 28, 238, 225, 110, 125, 244, 158, 250, 132, 247, 65, 28, 42, 113, 53, 79, 38, 104, 96, 209, 108, 150, 76, 219, 201, 173, 200, 115, 178, 12, 240, 213, 172, 0, 0, 0, 0] }))
[2022-08-23T14:11:47Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Arbitrating Lock final with 3 confirmations
[2022-08-23T14:11:47Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Refund not on the mempool
[2022-08-23T14:11:47Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Buy final with 1 confirmations
[2022-08-23T14:11:47Z DEBUG farcaster_node::swapd::runtime] request Buy tx task
[2022-08-23T14:11:48Z DEBUG farcaster_node::syncerd::bitcoin_syncer] creating transaction fetcher electrum client
[2022-08-23T14:11:49Z DEBUG farcaster_node::syncerd::bitcoin_syncer] successfully retrieved tx: "5865e62898cbff81d8248c3db3ca34f3ecc3c39f5256b0fb9513a94ced87e466"
[2022-08-23T14:11:49Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionRetrieved(TransactionRetrieved { id: TaskId(9), tx: Some(Transaction { version: 2, lock_time: 0, input: [TxIn { previous_output: OutPoint { txid: 361dc69c49393e3e58739ad4d1480ad03e5c7efb84c734494a06896320781df3, vout: 0 }, script_sig: Script(), sequence: 0, witness: Witness { content: [72, 48, 69, 2, 33, 0, 141, 179, 130, 157, 136, 124, 20, 245, 92, 0, 174, 216, 128, 37, 234, 189, 216, 9, 236, 208, 24, 125, 37, 149, 26, 24, 210, 225, 182, 141, 154, 75, 2, 32, 112, 28, 115, 52, 33, 155, 128, 156, 161, 73, 24, 157, 29, 139, 71, 66, 171, 149, 230, 9, 86, 76, 4, 186, 11, 233, 108, 60, 71, 174, 30, 215, 1, 72, 48, 69, 2, 33, 0, 189, 174, 180, 48, 239, 166, 38, 74, 4, 207, 48, 234, 103, 4, 63, 201, 72, 215, 122, 14, 3, 162, 33, 94, 75, 198, 137, 169, 28, 136, 51, 212, 2, 32, 13, 34, 249, 222, 112, 238, 167, 200, 130, 64, 158, 172, 0, 187, 5, 28, 169, 229, 9, 221, 56, 85, 9, 101, 71, 177, 194, 109, 167, 222, 20, 184, 1, 70, 33, 3, 39, 1, 85, 101, 47, 218, 34, 211, 101, 148, 64, 81, 148, 122, 210, 135, 27, 135, 104, 237, 65, 233, 248, 181, 64, 18, 53, 25, 20, 218, 45, 92, 173, 33, 2, 28, 238, 225, 110, 125, 244, 158, 250, 132, 247, 65, 28, 42, 113, 53, 79, 38, 104, 96, 209, 108, 150, 76, 219, 201, 173, 200, 115, 178, 12, 240, 213, 172], witness_elements: 3, last: 146, second_to_last: 73 } }], output: [TxOut { value: 1172, script_pubkey: Script(OP_0 OP_PUSHBYTES_20 9aaa50c17e6e3dbc7b3285e15e36d7e39353cdcd) }] }) }))
[2022-08-23T14:11:49Z WARN  farcaster_node::swapd::runtime] expected BobState(BuySigB), found BobState(CoreArb). Any chance you reused the destination/refund address in the cli command? For your own privacy, do not reuse bitcoin addresses. Txid 66e487ed4ca91395fbb056529fc3c3ecf334cab33d8c24d881ffcb9828e66558
[2022-08-23T14:12:08Z DEBUG farcaster_node::syncerd::monero_syncer] Watch wallet opened successfully
[2022-08-23T14:12:08Z DEBUG farcaster_node::syncerd::monero_syncer] FIXME: tx set to vec![0]
[2022-08-23T14:12:09Z DEBUG farcaster_node::syncerd::syncer_state] new tx seen: 9d01b8c00632b786c5c8d8f12c0d88f5c09a35ec3ff7afc03b6611ecfdf2d5ab
[2022-08-23T14:12:09Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, AddressTransaction(AddressTransaction { id: TaskId(7), hash: [157, 1, 184, 192, 6, 50, 183, 134, 197, 200, 216, 241, 44, 13, 136, 245, 192, 154, 53, 236, 63, 247, 175, 192, 59, 102, 17, 236, 253, 242, 213, 171], amount: 454116140000, block: [], tx: [0] }))
[2022-08-23T14:12:09Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Watching Accordant Lock transaction (9d01b8c00632b786c5c8d8f12c0d88f5c09a35ec3ff7afc03b6611ecfdf2d5ab)
[2022-08-23T14:12:09Z DEBUG farcaster_node::swapd::syncer_client] Watching transaction 9d01b8c00632b786c5c8d8f12c0d88f5c09a35ec3ff7afc03b6611ecfdf2d5ab with TaskId(10)
[2022-08-23T14:12:10Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(10), block: [15, 75, 36, 182, 252, 121, 118, 104, 132, 255, 30, 188, 204, 201, 224, 194, 10, 18, 153, 121, 165, 169, 80, 185, 178, 91, 67, 152, 152, 57, 226, 197], confirmations: Some(8), tx: [] }))
[2022-08-23T14:12:10Z INFO  farcaster_node::swapd::runtime] 0xc9e8…1ccf | State transition: BobState(CoreArb) -> BobState(BuySig)
[2022-08-23T14:12:11Z DEBUG farcaster_node::syncerd::monero_syncer] Watch wallet opened successfully
[2022-08-23T14:12:11Z DEBUG farcaster_node::syncerd::monero_syncer] FIXME: tx set to vec![0]
[2022-08-23T14:12:18Z INFO  farcaster_node::syncerd::syncer_state] Monero incremented height 1163766
[2022-08-23T14:12:18Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, HeightChanged(HeightChanged { id: TaskId(2), block: [207, 88, 158, 114, 96, 130, 106, 44, 112, 115, 91, 36, 45, 109, 116, 100, 84, 218, 156, 129, 90, 172, 36, 138, 55, 91, 248, 163, 236, 213, 155, 194], height: 1163766 }))
[2022-08-23T14:12:18Z DEBUG farcaster_node::swapd::syncer_client] Monero new height 1163766
[2022-08-23T14:12:19Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(10), block: [15, 75, 36, 182, 252, 121, 118, 104, 132, 255, 30, 188, 204, 201, 224, 194, 10, 18, 153, 121, 165, 169, 80, 185, 178, 91, 67, 152, 152, 57, 226, 197], confirmations: Some(9), tx: [] }))
[2022-08-23T14:12:19Z INFO  farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Accordant Lock final with 9 confirmations
[2022-08-23T14:15:21Z INFO  farcaster_node::syncerd::syncer_state] Monero incremented height 1163767
[2022-08-23T14:15:21Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, HeightChanged(HeightChanged { id: TaskId(2), block: [78, 30, 185, 85, 13, 31, 43, 13, 250, 124, 8, 116, 215, 98, 164, 56, 100, 135, 198, 36, 125, 215, 14, 213, 248, 158, 109, 182, 60, 71, 166, 100], height: 1163767 }))
[2022-08-23T14:15:21Z DEBUG farcaster_node::swapd::syncer_client] Monero new height 1163767
[2022-08-23T14:15:22Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, TransactionConfirmations(TransactionConfirmations { id: TaskId(10), block: [15, 75, 36, 182, 252, 121, 118, 104, 132, 255, 30, 188, 204, 201, 224, 194, 10, 18, 153, 121, 165, 169, 80, 185, 178, 91, 67, 152, 152, 57, 226, 197], confirmations: Some(10), tx: [] }))
[2022-08-23T14:15:22Z DEBUG farcaster_node::swapd::syncer_client] 0xc9e8…1ccf | Tx Accordant Lock final with 10 confirmations
[2022-08-23T14:15:31Z INFO  farcaster_node::syncerd::syncer_state] Monero incremented height 1163768
[2022-08-23T14:15:31Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0xc9e8…1ccf>, HeightChanged(HeightChanged { id: TaskId(2), block: [54, 154, 121, 72, 191, 107, 242, 204, 114, 11, 112, 248, 61, 40, 254, 45, 92, 2, 29, 84, 218, 171, 60, 85, 208, 41, 44, 252, 149, 165, 222, 236], height: 1163768 }))
```

**Comment by h4sh3d** [comment]
The problem happens because we restore `CoreArb` and wait on transition to `BuySig` that is triggered by seen the monero lock.

We get the monero lock from restored syncers, but we got it after the buy bitcoin transaction from the restored bitcoin syncer, so we are still in `CoreArb`, so we discard buy, and then later receive acc lock and transition to `BuySig`.
